plugins/woocommerce/src/Internal/EmailEditor/Integration.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\EmailEditor;
+
+use Automattic\WooCommerce\Internal\EmailEditor\Patterns\PatternsController;
+use Automattic\WooCommerce\Internal\EmailEditor\Templates\TemplatesController;
+
+defined( 'ABSPATH' ) || exit;
+
+class Integration {
+	const EMAIL_POST_TYPE = 'woo_mail';
+
+	private PageRenderer $editor_page_renderer;
+
+	public function init(): void {
+		$this->init_hooks();
+		$this->register_hooks();
+	}
+
+	/**
+	 * These classes have an register init method.
+	 */
+	public function init_hooks() {
+		$container = wc_get_container();
+		$container->get( PatternsController::class );
+		$container->get( TemplatesController::class );
+		$container->get( PersonalizationTagManager::class );
+		$this->editor_page_renderer = $container->get( PageRenderer::class );
+	}
+
+	public function register_hooks() {
+		add_filter('mailpoet_email_editor_post_types', [$this, 'add_email_post_type']);
+		add_filter('mailpoet_is_email_editor_page', [$this, 'is_editor_page'], 10, 1);
+		add_filter('replace_editor', [$this, 'replace_editor'], 10, 2);
+	}
+
+	public function add_email_post_type(array $postTypes): array {
+		$postTypes[] = [
+			'name' => self::EMAIL_POST_TYPE,
+			'args' => [
+				'labels' => [
+					'name' => __('Woo Emails', 'woocommerce'),
+					'singular_name' => __('Woo Email', 'woocommerce'),
+					'add_new_item' => __('Add New Woo Email', 'woocommerce'),
+					'edit_item' => __('Edit Woo Email', 'woocommerce'),
+					'new_item' => __('New Woo Email', 'woocommerce'),
+					'view_item' => __('View Woo Email', 'woocommerce'),
+					'search_items' => __('Search Woo Emails', 'woocommerce')
+				],
+				'rewrite' => ['slug' => self::EMAIL_POST_TYPE],
+				'supports' => ['title', 'editor'],
+				'public' => true,
```

Comment:
Do we want to use `public`? In MailPoet we use public set to `false` in combination with `publicly_queryable` because of previews. If I'm not mistaken with public `true` the email posts will also be searchable in the front end. So we could probably at least set `exclude_from_search` (see [docs](https://developer.wordpress.org/reference/functions/register_post_type/)). Perhaps there might be other effects.

Reply:
I'm keeping this public right now because
1. This is an initial implementation, and it is behind a feature flag; we can update it later
2. Using `public = false` or `exclude_from_search` leads to this error when saving the email. We need to take a closer look at it and fix it or update it later.
![Screenshot 2025-02-25 at 11 32 40 AM](https://github.com/user-attachments/assets/9fa99c10-55b6-4447-b630-6d7957b32044)

Reply:
Thanks for sharing the details. Could you please create an issue for investigating the error so that we don't forget? 

I think the error is related to this code https://github.com/woocommerce/woocommerce/blob/88f1c3c00c9bfc7c51f0b250623c4be761381d52/packages/php/email-editor/src/Engine/Templates/class-templates.php#L140-L167
If I'm not mistaken, we already added the templates registry so we could use it to list email templates instead of [fetching all templates and filtering by the prefix](https://github.com/woocommerce/woocommerce/blob/88f1c3c00c9bfc7c51f0b250623c4be761381d52/packages/php/email-editor/src/Engine/Templates/class-templates.php#L158-L166).

Reply:
I agree with you.
Done in https://github.com/woocommerce/woocommerce/issues/55862